xymol/xymol.py

```diff
@@ -133,3 +134,35 @@ def plot_similarity_map(
             self.mol, weights, **kwargs
         )
         img.savefig(file_name, bbox_inches='tight', dpi=600)
+
+    def create_map(
+        self, featurizer, model, filename="similarity_map.png"
+        ) -> list:
+        """Create similarity map
+
+        Args:
+            featurizer (object): must have a featurize() function.
+            model (object): must have a predict() function.
+
+        Returns:
+            list: a list of weights
+        """
+        if not( hasattr(featurizer, "featurize") and callable(getattr(featurizer, "featurize")) ):
+            raise ValueError("Featurizer must have featurize() function.")
```

Comment:
There should be at least one test on the wrong model and featurizer that do not have the corresponding APIs.